docs/standards/smart-contracts/lsp8-identifiable-digital-asset.md

```diff
@@ -335,6 +335,224 @@ _Triggers the **[Transfer](#trasnfer-2)** event when the tokens get successfully
+
+```solidity
+function _beforeTokenTransfer(
+    address from,
+    address to,
+    bytes32 tokenId
+) internal virtual
+```
+
+Hook that is called before any token transfer. This includes minting and burning.
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `tokenId` | bytes32  | The token to transfer. |
+
+:::note
+
+#### Requirements:
+
+- When `from` and `to` are both non-zero, ``from``'s `tokenId` will be transferred to `to`.
+- When `from` is zero, `tokenId` will be minted for `to`.
+- When `to` is zero, ``from``'s `tokenId` will be burned.
+
+:::
+
+### _notifyTokenSender
+
+```solidity
+function _notifyTokenSender(
+    address from,
+    address to,
+    bytes32 tokenId,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token sender about the `tokenId` changing owners using LSP1 interface.
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `tokenId` | bytes32  | The token to transfer. |
+| `data`   | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+### _notifyTokenReceiver
+
+```solidity
+function _notifyTokenReceiver(
+    address from,
+    address to,
+    bytes32 tokenId,
+    bool force,
+    bytes memory data
+) internal virtual
+```
+
+An attempt is made to notify the token receiver about the `tokenId` changing owners using LSP1 interface. When force is FALSE the token receiver MUST support LSP1.
+
+#### Parameters:
+
+|   Name   |   Type   | Description |
+| :------- | :------- | :---------- |
+| `from`   | address  | The sending address. |
+| `to`     | address  | The receiving address. |
+| `tokenId` | bytes32  | The token to transfer. |
+| `force`  | bool     | When set to TRUE, `to` may be any address; when set to FALSE `to` must be a contract that supports LSP1 UniversalReceiver and not revert. |
+| `memory` | bytes    | Additional data the caller wants included in the emitted event, and sent in the hooks to `from` and `to` addresses. |
+
+### _revokeOperator
+
+```solidity
+function _revokeOperator(
+    address operator,
+    address tokenOwner,
+    bytes32 tokenId
+) internal virtual
+```
+
+An attempt is made to notify the token receiver about the `tokenId` changing owners using LSP1 interface. When force is FALSE the token receiver MUST support LSP1.
+
+#### Parameters:
+
+|     Name     |   Type   | Description |
+| :----------- | :------- | :---------- |
+| `operator`   | address | The address to revoke as an operator.     |
+| `tokenOwner` | address | The address that is the owner of tokenId. |
+| `tokenId`    | bytes32 | The token to disable operator status to.  |
+
+### _clearOperators
+
+```
```

Comment:
please use ```solidity

Reply:
No description in the natspec, I added custom description.